### PR TITLE
[DUOS-2780] System Email - DAR submitted to DAC Chair

### PR DESCRIPTION
**Pull request by hams7504**
### Addresses
https://broadworkbench.atlassian.net/browse/DUOS-2780

### Summary
- Reformatted email template to add extra information
- Added new Dac DAO method that retrieves all dacs in a given collection

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/main/java/org/broadinstitute/consent/http/service/EmailService.java [hunk 1]
```diff
@@ -128,6 +150,26 @@ public List<MailMessage> fetchEmailMessagesByCreateDate(Date start, Date end, In
   public void sendNewDARCollectionMessage(Integer collectionId)
       throws IOException, TemplateException {
     DarCollection collection = collectionDAO.findDARCollectionByCollectionId(collectionId);
+    List<User> distinctUsers = getDistinctUsers(collection);
+    DataAccessRequest dar = dataAccessRequestDAO.findByReferenceId(collection.getDarCode());
```
Comment (rushtong) [inline src/main/java/org/broadinstitute/consent/http/service/EmailService.java]: We need to fix this, the dar code is not the same as the reference id.

File: src/main/java/org/broadinstitute/consent/http/db/DacDAO.java [hunk 2]
```diff
@@ -219,4 +220,15 @@ WHERE ds.dataset_id IN (<datasetIds>)
...
 
+  @RegisterRowMapper(DacMapper.class)
+  @SqlQuery("""
+      select dac.* 
+      from dac
+      inner join dataset d on d.dac_id = dac.dac_id
+      inner join dar_dataset dd on dd.dataset_id = d.dataset_id
+      inner join data_access_request dar on dd.reference_id = dar.reference_id
+      where dar.collection_id = :collectionId
+      """)
```
Comment (rushtong) [inline src/main/java/org/broadinstitute/consent/http/db/DacDAO.java]: We should standardize the casing here

File: src/main/java/org/broadinstitute/consent/http/service/EmailService.java [hunk 3]
```diff
@@ -128,6 +150,26 @@ public List<MailMessage> fetchEmailMessagesByCreateDate(Date start, Date end, In
...
+    String researcherName = userDAO.findUserById(dar.getUserId()).getDisplayName();
+    Collection<Dac> dacsInDAR = dacDAO.findDacsForCollectionId(collectionId);
+    List<Dataset> datasetsInDAR = datasetDAO.findDatasetsByIdList(dar.datasetIds);
```
Comment (rushtong) [inline src/main/java/org/broadinstitute/consent/http/service/EmailService.java]: We need some safety checking around the bulk calls to ensure we're not passing empty lists.

File: src/main/java/org/broadinstitute/consent/http/service/EmailService.java [hunk 4]
```diff
@@ -128,6 +150,26 @@ public List<MailMessage> fetchEmailMessagesByCreateDate(Date start, Date end, In
...
+    for (User user : distinctUsers) {
+      List<Dac> matchingDacsForUser = getMatchingDacs(user, dacsInDAR);
+      for (Dac dac : matchingDacsForUser) {
+        List<String> matchingDatasetsForDac = getMatchingDatasets(dac, datasetsInDAR);
+        if (matchingDatasetsForDac != null) {
+          sendList.put(dac.getName(), matchingDatasetsForDac);
+        }
+      }
+      sendNewDARRequestEmail(user, sendList, researcherName, collection.getDarCode());
+    }
```
Comment (rushtong) [inline src/main/java/org/broadinstitute/consent/http/service/EmailService.java:174]: We should try to flatten this nested loop

**Review by rushtong — COMMENTED** [review]
Going to tackle this feedback myself.

File: src/main/java/org/broadinstitute/consent/http/service/EmailService.java (hunk 4 above)
Comment (rushtong) [reply src/main/java/org/broadinstitute/consent/http/service/EmailService.java:174]: I don't think we can in this case.

**Review by rushtong — APPROVED** [review]
Manually tested 👍🏽

**Review by Shakespeared — APPROVED** [review]
seems reasonable, lgtm but didn't manually test

**Pull request merged**